scipy/linalg/_decomp.py

```diff
@@ -123,6 +124,12 @@ def eig(a, b=None, left=False, right=True, overwrite_a=False,
 
     where ``.H`` is the Hermitian conjugation.
 
+    .. note::
+
+        The documentation is written assuming array arguments of are of specified
+        shapes. However, array argument(s) of this function may be N-dimensional, in
+        which case they follow  *standard rules for batched linear algebra operations*.
```

Comment:
This doesn't need to be added manually.

Reply:
I think I'd prefer regular updates of the parameter description over this note. As an example, here is what [torch.linalg.eig](https://pytorch.org/docs/main/generated/torch.linalg.eig.html) uses for the first input parameter:

_A ([Tensor](https://pytorch.org/docs/main/tensors.html#torch.Tensor)) – tensor of shape (*, n, n) where * is zero or more batch dimensions consisting of diagonalizable matrices._

That will read nicer (it's just a feature, not a special case treated with a `.. note::` admonition), and also gives the space to clarify behavior better. For example for this function: if `a` has batch dimensions, must `b` have them too with matching shape? What happens if `b` isn't batched, does it broadcast or raise?

Reply:
The matrices are not necessarily diagonalizable though. Not sure why they added that detail. It can have jordan blocks, (not necessarily computed accurately or computed at all but it does not require diagonalizability to be a tensor).

Reply:
> For example for this function: if a has batch dimensions, must b have them too with matching shape? What happens if b isn't batched, does it broadcast or raise?

For all functions, the decorator broadcasts the  batch shapes, ignoring the core dimensions. The concepts here would be covered in the linked tutorial. 

> That will read nicer

It will read *nicer* than a description in terms of only the core dimensions?

If this is very important, I will open a separate issue to discuss because there's a bit more to be said.

Of course, I would be happy to change the location of a standard note.

Let me know if I should open that issue.

Reply:
I took the note out of the admonition and made it plain text, and I linked to the appropriate part of the `linalg` tutorial. 

![image](https://github.com/user-attachments/assets/0b68ba57-9ef3-487f-900d-96db08a76dbd)

I drafted the part of the tutorial I envisioned:

https://github.com/scipy/scipy/blob/1c21193035bc3563cde13a45656375e1c22461c9/doc/source/tutorial/linalg_batch.ipynb

When it looks good, I'll convert it to markdown and link to it from the tutorial and the standard note. With this plan, the documentation identifies the *core shape*s  of the inputs/outputs and is written for a *batch shape* of `()`. This is (much) easier for us, less error-prone, and it is as simple as possible for readers who are interested in the typical use case. For the rest, it is easy to extend to higher-dimensional batches according to the the general rules given in the tutorial.